### DISCUSSION TITLE
Elevation gained should be included in run data

### DISCUSSION THREAD

**Discussion by irvingbennett** [yihong0618/running_page#509] [General]
Hello, 
For trail runners, elevation gained per run, and accumulated, is an important stat. It should be easy to include that datum in the yearly stats. I will try to include it in my running page. I am sure that those who know the code well could do it much faster than I could do it. Any help would be appreciated. My guess is that I would need to modify several parts to get it all to work. In Python I would need to add that column to the Pandas dataframe, grab the info from the activity and include it. Then, in the loop that aggregates the data, it needs to be summed up. Lastly, the React Component that displays it needs to modified as well. Is this something that could be of general interest?
Cheers!
Irving
![Strava_Run_Climbing_Challenge_September_2023](https://github.com/yihong0618/running_page/assets/3063697/83709be0-2a89-4ec5-b663-29821075f37c)

**Comment by yihong0618** [comment]
@irvingbennett Hi sorry for late reply
I think you are right, Elevation is import, we also consider to add it
the hard part is:

- we need do some migrate for the sqlite database, for early data
- the following as you mentioned
- and we also should consider the table part if we add elevation, the length maybe lager and not well display maybe

we may add it for an option later thank you for the advice

@ben-29  @frostming @shaonianche what do you think?

**Reply by ben-29** [reply]
For SQLite migration, I think we can add the `elevation_gain` column directly. As for old data, The easiest way is to let the users do a full reimport if they're interested in `elevation_gain`.
For table display, we may try to add the elevation_gain column or add it to the title of the detail map.

**Comment by ben-29** [comment]
Hi Irving,

I've added the Elevation Gain field. Could you please test it out with PR #719?
